haskell/haddock.bzl

```diff
@@ -323,6 +323,7 @@ def _haskell_doc_rule_impl(ctx):
 
     return [DefaultInfo(
         files = depset(html_dict_copied.values() + [index_root]),
+        runfiles = ctx.runfiles(html_dict_copied.values() + [index_root]),
```

Comment:
The `html_dict` is huge and here we are building two temporaries copies of it.

I do wonder what can be the impact on performance. Something such as:

```
    l = html_dict_copied.values() + [index_root]
    return [DefaultInfo(
￼        files = depset(l),
￼        runfiles = ctx.runfiles(l),
```

may be better.

Reply:
Good catch. It's tracking directories, not individual files, so one entry per target, but you're right, better not copy it unnecessarily.